dio/lib/src/adapters/io_adapter.dart

```diff
@@ -7,6 +7,8 @@ import '../dio_error.dart';
 import '../redirect_record.dart';
 
 typedef OnHttpClientCreate = HttpClient? Function(HttpClient client);
+typedef ResponseCertApprover = bool Function(
```

Comment:
The function only represents a boolean value. I would prefer `CertificatePredicate`.

Reply:
I took a deep dive in reaction to this question. Naming is important, and I think it deserves consideration. I looked at synonyms and definitions. In the end, I think we should change the typedef name, but I think we should follow existing Dio naming conventions. Consistency within a library is also incredibly important, in my opinion.

The [`RequestOptions` class](https://pub.dev/documentation/dio/latest/dio/RequestOptions-class.html) has a function that you can provide to check that the HTTP status code is acceptable. It's called [`ValidateStatus`](https://pub.dev/documentation/dio/latest/dio/ValidateStatus.html).

I think we should rename this to `ValidateCertificate` or maybe `ValidateResponseCertificate`.

The word "predicate" [has the precise definition](https://dcl-prog.stanford.edu/function-predicate.html) you mention, but functional programming is still niche, and sometimes intimidating. Rx-style work has widened that group, but the majority of developers will likely see it as a technical term that doesn't reflect the role this function has. (Even `CertificatePredicate` is too vague: does this predicate tell me if the last bit is a 1 or a 0? does it tell me if the host name is contained within the certificate?) The name should reflect the role it plays. I think `ValidateCertificate` conveys that role, implies a yes-or-no outcome, and provides parallelism with `ValidateStatus`.

What do you think? 😃

Reply:
I agree with `ValidateCertificate` from your perspetive, and thanks for the rich context!

Reply:
Excellent, I'll contact you about the timing of this. Right now I'm thinking I can do it in about 2 weeks so we can migrate our dependencies internally.

Reply:
@jgoyvaerts Is there any chance that PRs like these will be merged? I see that #1590 was just recently merged. Lack of activity on this library is making a number of people nervous. Can we expect to see more PRs merging in the days ahead?